api/services/v1alpha1/monitoring_types.go

```diff
@@ -41,27 +41,27 @@ type MonitoringSpec struct {
 
 // Metrics defines the desired state of metrics for the monitoring service
 type Metrics struct {
-	Storage   MetricsStorage   `json:"storage,omitempty"`
-	Resources MetricsResources `json:"resources,omitempty"`
+	Storage   *MetricsStorage   `json:"storage,omitempty"`
+	Resources *MetricsResources `json:"resources,omitempty"`
 }
 
-// MetricsStorage defines the desired state of storage for the monitoring service
+// MetricsStorage defines the storage configuration for the monitoring service
 type MetricsStorage struct {
-	// Size of the storage in Gi
-	Size int `json:"size,omitempty"`
-	// Retention of the storage in days
-	Retention int `json:"retention,omitempty"`
+	// Size specifies the storage size for the MonitoringStack (e.g, "5Gi", "10Mi")
+	Size string `json:"size,omitempty"`
+	// Retention specifies how long metrics data should be retained (e.g., "1d", "2w")
+	Retention string `json:"retention,omitempty"`
 }
 
-// MetricsResources defines the desired state of resource requests and limits for the monitoring service
+// MetricsResources defines the resource requests and limits for the monitoring service
 type MetricsResources struct {
```

Comment:
For quantities, there is a `resource.Quantity` type, have we thought about using it ?